engine/src/mac-window.mm

```diff
@@ -2246,20 +2251,29 @@ void MCMacPlatformHandleDrawSync(NSWindow *window)
 		// The timeout value of 0.02ms is specified to avoid hitting the 60hz redraw limit.
 		if (!s_inside_focus_event && !s_showing_sheet && ![m_delegate inUserReshape])
 		{
-			m_waiting_for_draw = true;
+			NSMutableArray *t_popped_events = [NSMutableArray array];
 			while (m_waiting_for_draw)
 			{
 				NSEvent *t_event;
 				t_event = [NSApp nextEventMatchingMask: NSApplicationDefinedMask
 											 untilDate: [NSDate dateWithTimeIntervalSinceNow: 0.02]
 												inMode: NSEventTrackingRunLoopMode
-											   dequeue: NO];
+											   dequeue: YES];
+				if (t_event != nil && !MCMacPlatformIsDrawSyncEvent(t_event))
+					[t_popped_events insertObject:t_event atIndex:t_popped_events.count];
```

Comment:
I think `NSMutableArray` has an `addObject` member which does this (inserts at the end).

Reply:
yep, have updated this now :)